packages/gatsby-plugin-sitemap/src/gatsby-node.js

```diff
@@ -26,7 +40,37 @@ exports.onPostBuild = async ({ graphql, pathPrefix }, pluginOptions) => {
     excludeOptions,
     pathPrefix
   )
-  serialize(queryRecords).forEach(u => map.add(u))
+  const urls = serialize(queryRecords)
 
-  return await writeFile(saved, map.toString())
+  if (!(rest.sitemapSize && urls.length > rest.sitemapSize)) {
```

Comment:
I have a hard time wrapping my head around `!(conditions)` probably just me.

```suggestion
  if (!rest.sitemapSize || urls.length <= rest.sitemapSize) {
```